### ISSUE TITLE
Feature Request: Mock Support for AVPlayerPlaybackCoordinator.coordinateWithSession

### ISSUE THREAD

**Issue by PedroRossa** [Pixeland-Tech/SharePlayMock#10]
I’m currently using SharePlayMock to test SharePlay functionality in my visionOS project, which heavily relies on coordinated media playback using AVPlayerPlaybackCoordinator. However, I noticed that SharePlayMock does not currently support mocking the coordinateWithSession method of AVPlayerPlaybackCoordinator, which is crucial for testing coordinated playback scenarios with SharePlay.

## Feature Suggestion

It would be highly beneficial if SharePlayMock could implement support for AVPlayerPlaybackCoordinator.coordinateWithSession. This feature would enable developers to test coordinated playback behavior in a controlled environment, closely mimicking the actual SharePlay experience.

## Use Case
- Synchronizing video playback states between participants in a SharePlay session.
- Testing and validating playback coordination logic during SharePlay interactions.
- Simulating various states of the playback coordinator, such as starting, pausing, and seeking, to ensure consistent behavior across all participants.

## Expected Behavior
- The mock should allow AVPlayerPlaybackCoordinator to coordinate with the mocked GroupSession, enabling the developer to test playback synchronization between participants.
- It should be able to simulate participant join/leave scenarios, changes in playback state, and other session dynamics that influence coordinated playback.

## Benefits
- Enhances the utility of SharePlayMock by covering a key aspect of SharePlay development.
- Allows developers to perform more comprehensive testing of SharePlay-integrated apps without needing a live SharePlay session.

Thank you for considering this feature request! Your work on SharePlayMock has already been incredibly helpful, and adding support for playback coordination would further extend its capabilities.

**Comment by xinyic** [comment]
Hi Pedro, thank you for submitting the feature request! We took a look at AVPlayerPlaybackCoordinator and coordinateWithSession. We found that coordinateWithSession's argument type is GroupSession and decide that we could not support mocking AVPlayerPlaybackCoordinator in the near future. The reasoning is

coordinateWithSession's argument type is GroupSession class instead of a protocol
=> meaning that we can't just have GroupSessionMock conform that protocol
=> meaning that we have to mock the whole AVPlaybackCoordinator class (so that coordinateWithSession can take in GroupSessionMock as an argument) and this class is complicated and will need some time and effort to build the mock function

If you'd like to implement that mock or have another suggestion on how to implement it, we'd be more than happy to merge/discuss it :)
